pkg/controller/discovery/web/ns.go

```diff
@@ -117,29 +167,107 @@ func (h NsHandler) List(ctx *gin.Context) {
 //
 // Get a specific namespace on a cluster.
 func (h NsHandler) Get(ctx *gin.Context) {
-	status := h.Prepare(ctx)
-	if status != http.StatusOK {
-		ctx.Status(status)
-		return
+	ctx.Status(http.StatusMethodNotAllowed)
+}
+
+//
+// Token has MIG-admin.
+func (h *NsHandler) hasAdmin() (bool, error) {
+	return h.rbac.Allow(&auth.Review{
+		Group:     migapi.SchemeGroupVersion.Group,
+		Namespace: auth.ALL,
+		Resource:  auth.ALL,
+		Verb:      auth.ALL,
+	})
+}
+
+//
+// Filter the list of namespaces based on token having
+// at least READ access.
+func (h *NsHandler) hasRead(models []*model.Namespace) ([]*model.Namespace, error) {
+	hasAdmin, err := h.hasAdmin()
```

Comment:
can we please put a comment before this function? IMO it will help in re-reading this code in future
```
// check and return fast if user has access to mig-api in all namespaces
```